app/src/Grout.js

```diff
@@ -0,0 +1,273 @@
+
+    getFilters(uuid) {
+        /*
+         * Returns the filterable fields for a RecordSchema given its UUID.
+         *
+         * @params {string} uuid - The UUID for the RecordSchema whose fields
+         *                         should be returned.
+         * @returns {Promise} - If the query is successful, returns an array
+         *                      of fields that can be filtered. Otherwise, throws
+         *                      an error.
+         */
+        return new Promise((resolve, reject) => {
+            this.get(uuid)
+                .then(schema => {
+                    const filters = this._parseFilters(schema);
+                    resolve(filters);
+                })
+                .catch(error => reject(error));
+        });
+    }
+
+    _parseFilters(schema) {
+        /*
+         * Private helper method to parse a set of filterable fields given
+         * a RecordSchema object.
+         */
+        // Each RecordSchema stores its field specifications in a nested object
+        // inside of the `schema.definitions` object, so check to make sure that
+        // those definitions exist. If they don't, the object will be empty,
+        // and most of the logic of this method will be skipped.
+        let definitions = {};
+        if (schema.schema && schema.schema.definitions) {
+            definitions = schema.schema.definitions;
+        }
+
+        let schemaFilters = {};
+
+        // The field definitions are mapped to a form name, to allow admins to
+        // create multiple forms for a given schema (the default form is
+        // "${RecordType.label}Details"). Since there can be multiple forms in
+        // a given schema, iterate the keys of the field definitions object in
+        // order to extract fields from each form.
+        Object.keys(definitions).forEach(key => {
+            let schema = definitions[key];
+            schemaFilters[schema.title] = {};
+
+            // The `schema.properties` object stores the fields on this form.
+            Object.keys(schema.properties).forEach((prop) => {
+                let property = schema.properties[prop];
+                // The `isSearchable` flag determines whether the field can
+                // be filtered or not.
+                if (property.isSearchable) {
+                    // Merge some information in with the properties to
+                    // facilitate queries later on.
+                    let propertyAdditions = {
+                        multiple: schema.multiple,  // The containment type.
+                        field: prop  // The name of the field.
+                    }
+                    let formattedProperty = Object.assign({}, property, propertyAdditions);
+
+                    // Namespace the filter IDs with `key + # + prop`, in case
+                    // multiple forms have the same name for fields.
+                    schemaFilters[schema.title][key + '#' + prop] = formattedProperty;
```

Comment:
I'm not very up to speed on how the original AngularJS code works right now, but I do remember it being one of the most complicated and least satisfactory parts of DRIVER, so my inclination is that if you think there's a better way, there probably is. I think the way you've designed it here is reasonable, but just to push on this a bit, what do you think about storing the filters as an array of objects? It would be more duplicative of the information about the path to each field, but it might make generating interface elements from each filter easier since they would all be completely self-contained.